web/hooks/use-is-visible.ts

```diff
@@ -0,0 +1,31 @@
+import { useEffect, useRef } from 'react'
+
+export function useIsVisible(cb: () => void, once = false) {
+  const ref = useRef<HTMLDivElement | null>(null)
+  useEffect(() => {
+    const element = ref.current
+    if (!element) return
+
+    const observer = new IntersectionObserver(
+      (entries) => {
+        entries.forEach((entry) => {
+          if (entry.isIntersecting) {
+            cb()
+            if (once) {
```

Comment:
We might not need this once bit or we might want to track how long the card was viewed for.

Reply:
Yup. Seems good for now though